:memo: Update Mask Generation Task

rajaryan18 opened the pull request:
Added about.md for Zero-shot-object-detection, text-to-3d and mask-generation. Also added use cases for them all.

merveenoyan commented on packages/tasks/src/tasks/mask-generation/about.md:
```diff
@@ -0,0 +1,7 @@
+## About the Task
```
```suggestion
```
No need to add this header, it's already added in `summary` in `data.ts`

merveenoyan commented on packages/tasks/src/tasks/mask-generation/about.md:
```diff
@@ -0,0 +1,7 @@
+## About the Task
+
+Mask Generation task involves generating masks/segments for objects in a given image. It could be zero shot or based on training data.
```
```suggestion
Mask generation task involves generating masks for meaningful segments in a given image. It could be zero-shot or based on training data.
```

merveenoyan commented on packages/tasks/src/tasks/mask-generation/about.md:
```diff
@@ -0,0 +1,7 @@
+
+## Use Cases
+
+Mask Generation has promising use cases in Medical Imaging, Autonomous Vehicles, Robotics, etc.
```
In this section we expect to have multiple headers for each use case.

merveenoyan commented on packages/tasks/src/tasks/zero-shot-object-detection/about.md:
```diff
@@ -0,0 +1,7 @@
+## About the Task
```
This page is already contributed in another PR, so if you feel like contributing it, you can open PR to that branch as that one is more extensive

merveenoyan commented on packages/tasks/src/tasks/text-to-3d/about.md:
```diff
@@ -0,0 +1,7 @@
+## About the Task
+
+Text to 3D is a Generative task of generating a 3d image from a given text prompt. It is similar to generating a 2d image but involves an extended complexity of depth perception.
```
You can add this in `summary` in `data.ts`

```suggestion
Text-to-3D is to generate a 3D image from a text prompt. It is similar to generating a 2D image but involves an extended complexity of depth perception.
```

merveenoyan commented on packages/tasks/src/tasks/text-to-3d/about.md:
```diff
@@ -0,0 +1,7 @@
+## About the Task
```
No need to add, we add this in `summary`

merveenoyan commented on packages/tasks/src/tasks/text-to-3d/about.md:
```diff
@@ -0,0 +1,7 @@
+
+## Use Cases
+
+Text to 3d has various creative and experimental use cases like setting up virtual experiments and simulations.
```
Same comment applies here

merveenoyan reviewed (commented):
Thanks a lot for this PR! You can add your HF user handle to end of `about.md`.

Moreover, feel free to remove the empty files if you won't contribute content inside.

rajaryan18 commented:
@merveenoyan 
Hey, I have added the changes you suggested. Thank you so much for your time.

merveenoyan commented on .vscode/settings.json:
```diff
@@ -2,7 +2,7 @@
 	"editor.defaultFormatter": "esbenp.prettier-vscode",
```
Can you remove this file altogether?

merveenoyan commented on packages/tasks/src/tasks/mask-generation/about.md:
```diff
@@ -0,0 +1,11 @@
+
+### Medical Imaging
+
+Mask Generation is used in medical imaging in aiding in the segmentation and analysis of specific regions.
```
```suggestion
Mask generation models are used in medical imaging to aid in segmenting and analyzing specific regions.
```

merveenoyan commented on packages/tasks/src/tasks/mask-generation/about.md:
```diff
@@ -0,0 +1,11 @@
+
+### Autonomous Vehicles
+
+Mask Generation is used to create segments and masks for obstacles and other objects in view.
```
```suggestion
Mask generation models are used to create segments and masks for obstacles and other objects in view.
```

merveenoyan commented on packages/tasks/src/tasks/mask-generation/about.md:
```diff
@@ -0,0 +1,11 @@
+
+Mask Generation is used to create segments and masks for obstacles and other objects in view.
+
+This page was written by [Raj Aryan](https://huggingface.co/thatrajaryan)
```
```suggestion
This page was made possible thanks to the efforts of [Raj Aryan](https://huggingface.co/thatrajaryan).
```

merveenoyan commented on packages/tasks/src/tasks/text-to-3d/data.ts:
```diff
@@ -0,0 +1,30 @@
+	],
+	demo: {
+		inputs: [
+			
```
can you remove these gaps in the empty lists?

merveenoyan commented on packages/tasks/src/tasks/text-to-3d/data.ts:
```diff
@@ -0,0 +1,30 @@
+	datasets: [
+		{
+			description: "3D-Future is a dataset designed to generate 3D images from textual descriptions.",
+			id: "3D-Future",
```
This dataset isn't on Hugging Face Hub, perhaps you can create the dataset? would be great apparently there's none on Hub and you'd be the first person to add such dataset!

merveenoyan commented on packages/tasks/src/tasks/text-to-3d/data.ts:
```diff
@@ -0,0 +1,30 @@
+	spaces: [
+		
+	],
+	summary:"Text-to-3D is to generate a 3D image from a text prompt. It is similar to generating a 2D image but involves an extended complexity of depth perception.",
```
```suggestion
	summary: "Text-to-3D is the task of generating a 3D image from a text prompt. It is similar to generating a 2D image with an extension of depth perception.",
```

merveenoyan reviewed (commented):
Thanks a lot!

rajaryan18 commented:
@merveenoyan 
Hey, I have made the changes.
Changed .vscode/setting.json back to its original content (from this repository).
Removed the dataset 3D-Future from data.ts. Will find it and add to the hub and then maybe modify this file again to mention it.

merveenoyan commented on .vscode/settings.json:
```diff
@@ -7,4 +7,4 @@
 		"editor.defaultFormatter": "esbenp.prettier-vscode"
 	}
-}
+}
```
there seems to be a change still (see diff), if you can revert this change as well we can merge

merveenoyan reviewed (approved):
Thank you!

rajaryan18 commented:
@merveenoyan 
Hi again, I have changed that file. If it still crashes the checks, I'll just open a new PR and ensure to not commit any changes to .vscode folder.
Thank you

merveenoyan commented:
@rajaryan18 can you merge this PR into your branch? https://github.com/rajaryan18/huggingface.js/pull/1 this will lint your code so the tests pass

pcuenca commented on packages/tasks/src/tasks/mask-generation/data.ts:
```diff
@@ -1,7 +1,16 @@
 const taskData: TaskDataCustom = {
-	datasets: [],
+	datasets: [
+		{
+			description: "Widely used benchmark dataset for multiple Vision tasks.",
+			id: "merve/coco2017",
+		},
+        {
+            description: "Medical Imaging dataset of the Human Brain for segmentation and mask generating tasks",
+            id: "rocky93/BraTS_segmentation"
+        }
```
```suggestion
		{
			description: "Widely used benchmark dataset for multiple Vision tasks.",
			id: "merve/coco2017",
		},
                {
                    description: "Medical Imaging dataset of the Human Brain for segmentation and mask generating tasks",
                    id: "rocky93/BraTS_segmentation"
                }
```

pcuenca commented on packages/tasks/src/tasks/mask-generation/data.ts:
```diff
@@ -16,7 +25,13 @@ const taskData: TaskDataCustom = {
-	metrics: [],
+	metrics: [
+		{
+			description:
+				"IoU is used to measure the overlap between predicted mask and the ground truth mask.",
```
```suggestion
			description: "IoU is used to measure the overlap between predicted mask and the ground truth mask.",
```

pcuenca commented on packages/tasks/src/tasks/mask-generation/data.ts:
```diff
@@ -52,4 +67,4 @@ const taskData: TaskDataCustom = {
 };
 
-export default taskData;
+export default taskData;
```
```suggestion
export default taskData;

```

pcuenca commented on packages/tasks/src/tasks/mask-generation/data.ts:
```diff
@@ -1,7 +1,16 @@
 
 const taskData: TaskDataCustom = {
-	datasets: [],
+	datasets: [
+		{
+			description: "Widely used benchmark dataset for multiple Vision tasks.",
+			id: "merve/coco2017",
+		},
+                {
+                    description: "Medical Imaging dataset of the Human Brain for segmentation and mask generating tasks",
+                    id: "rocky93/BraTS_segmentation"
+                }
+	],
```
```suggestion
	datasets: [
		{
			description: "Widely used benchmark dataset for multiple Vision tasks.",
			id: "merve/coco2017",
		},
		{
			 description: "Medical Imaging dataset of the Human Brain for segmentation and mask generating tasks",
			 id: "rocky93/BraTS_segmentation"
		},
	],
```

pcuenca commented on packages/tasks/src/tasks/mask-generation/data.ts:
```diff
@@ -16,7 +25,12 @@ const taskData: TaskDataCustom = {
+		{
+			description: "IoU is used to measure the overlap between predicted mask and the ground truth mask.",
+			id: "Intersection over Union (IoU)",
+		}
```
```suggestion
		},
```

pcuenca commented on packages/tasks/src/tasks/mask-generation/data.ts:
```diff
@@ -1,7 +1,16 @@
 
 const taskData: TaskDataCustom = {
-	datasets: [],
+	datasets: [
+		{
+			description: "Widely used benchmark dataset for multiple Vision tasks.",
+			id: "merve/coco2017",
+		},
+		{
+			 description: "Medical Imaging dataset of the Human Brain for segmentation and mask generating tasks",
+			 id: "rocky93/BraTS_segmentation"
+		},
```
```suggestion
	datasets: [
		{
			description: "Widely used benchmark dataset for multiple Vision tasks.",
			id: "merve/coco2017",
		},
		{
			description: "Medical Imaging dataset of the Human Brain for segmentation and mask generating tasks",
			id: "rocky93/BraTS_segmentation",
		},
```

pcuenca commented:
I fixed the conflicts and brought this up to date, but the linter still fails for some reason cc @coyotte508 @merveenoyan

pcuenca commented:
Thank you @coyotte508! Merging now, thanks @rajaryan18 for the contribution and sorry for the delay 🙌
